feat(react): make session requireable in `useSession` (#2236)
nextauthjs/next-auth: www/docs/getting-started/example.md

You can use the `useSession` hook from anywhere in your application (e.g. in a header component).
:::

### Share/configure session state

> Use the `<SessionProvider>` to allows instances of `useSession()` to share the session object across components. It also takes care of keeping the session updated and synced between tabs/windows.

Reviewer: ```suggestion
Use the `<SessionProvider>` to allow instances of `useSession()` to share the session data across components. `<SessionProvider>` will also take care of keeping the session updated and synced between tabs/windows.
```
I'm not sure about this though given is not possible anymore to use `useSession` without `<SessionProvider />` 🤔

Author: Yes, we should rephrase this to be explicit that users *must* use `SessionProvider` from now on.
